Content/Tiles/Cooking/Salts.cs

```diff
@@ -0,0 +1,289 @@
+		{
+			if (Main.tile[i, j].LiquidAmount > 0)
+				WorldGen.KillTile(i, j, default, default, true);
+		}
+
+		public override bool TileFrame(int i, int j, ref bool resetFrame, ref bool noBreak)
+		{
+			if (WorldGen.noTileActions)
+				return true;
+
+			Tile above = Main.tile[i, j - 1];
+			Tile below = Main.tile[i, j + 1];
+			bool canFall = true;
+
+			if (below == null || below.HasTile)
+				canFall = false;
+
+			if (above.HasTile && (TileID.Sets.BasicChest[above.TileType] || TileID.Sets.BasicChestFake[above.TileType] || above.TileType == TileID.PalmTree || TileID.Sets.BasicDresser[above.TileType]))
+				canFall = false;
+
+			if (canFall)
+			{
+				float positionX = i * 16 + 8;
+				float positionY = j * 16 + 8;
+
+				if (Main.netMode == NetmodeID.SinglePlayer)
+				{
+					Main.tile[i, j].ClearTile();
+					int proj = Projectile.NewProjectile(null, positionX, positionY, 0f, 0.41f, ProjectileType, 10, 0f, Main.myPlayer);
+					Main.projectile[proj].ai[0] = 1f;
+					WorldGen.SquareTileFrame(i, j);
+				}
+				else if (Main.netMode == NetmodeID.Server)
+				{
+					Main.tile[i, j].ClearTile();
+					bool spawnProj = true;
+
+					for (int k = 0; k < 1000; k++)
+					{
+						Projectile otherProj = Main.projectile[k];
+
+						if (otherProj.active && otherProj.owner == Main.myPlayer && otherProj.type == ProjectileType && Math.Abs(otherProj.timeLeft - 3600) < 60 && otherProj.Distance(new Vector2(positionX, positionY)) < 4f)
+						{
+							spawnProj = false;
+							break;
+						}
+					}
+
+					if (spawnProj)
+					{
+						int proj = Projectile.NewProjectile(null, positionX, positionY, 0f, 2.5f, ProjectileType, 10, 0f, Main.myPlayer);
+						Main.projectile[proj].velocity.Y = 0.5f;
+						Main.projectile[proj].position.Y += 2f;
+						Main.projectile[proj].netUpdate = true;
+					}
+
+					NetMessage.SendTileSquare(-1, i, j, 1);
+					WorldGen.SquareTileFrame(i, j);
+				}
+
+				return false;
+			}
+
+			return true;
+		}
+	}
+
+	internal class TableSaltProjectile : ModProjectile
+	{
+		protected bool falling = true;
+
+		protected virtual int TileType => ModContent.TileType<TableSalt>();
+
+		public virtual int DustType => DustID.Marble;
+
+		public override string Texture => AssetDirectory.CookingTile + Name;
+
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Sea Salt");
+			ProjectileID.Sets.ForcePlateDetection[Projectile.type] = true;
+		}
+
+		public override void SetDefaults()
+		{
+			Projectile.knockBack = 6f;
+			Projectile.width = 10;
+			Projectile.height = 10;
+			Projectile.friendly = true;
+			Projectile.hostile = true;
+			Projectile.penetrate = -1;
+		}
+
+		public override void AI()
+		{
+			if (Main.rand.NextBool(5))
+			{
+				int dust = Dust.NewDust(Projectile.position, Projectile.width, Projectile.height, DustType);
+				Main.dust[dust].velocity.X *= 0.4f;
+			}
+
+			Projectile.tileCollide = true;
+			Projectile.localAI[1] = 0f;
```

Comment:
We should be creating ref properties to give these fields readable names, Additionally we should use custom variables as opposed to localAI as localAI provides no benefits